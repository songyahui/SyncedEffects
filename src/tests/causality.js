"use hopscript"

var hh = require("hiphop");

var example = <hh.module I O>
  <hh.loop>
    <hh.sequence>
      <hh.if O>
	<hh.emit I/>
      </hh.if>
      <hh.pause/>
      <hh.emit O/>
    </hh.sequence>
  </hh.loop>
</hh.module>;

exports.prg = new hh.ReactiveMachine(example, "presentemit");
