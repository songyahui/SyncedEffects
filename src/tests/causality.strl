module presentemit:
output I, O;
loop
    present O then
        emit I
    end present;
    pause;
    emit O
end loop
end module
