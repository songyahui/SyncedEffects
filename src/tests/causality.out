presentemit> ;
--- Output:
presentemit> ;
--- Output: I O
presentemit> ;
--- Output: I O
presentemit> ;
--- Output: I O
